src/dev/so_migration/snapshot_plugin_types.ts

```diff
@@ -10,30 +10,15 @@ import * as fs from 'fs';
 import * as path from 'path';
 import * as cp from 'child_process';
 
-import {
-  extractMigrationInfo,
-  getMigrationHash,
-  SavedObjectTypeMigrationInfo,
-  // TODO: how to resolve this? Where to place this script?
-  // eslint-disable-next-line @kbn/imports/no_boundary_crossing
-} from '@kbn/core-test-helpers-so-type-serializer';
-import {
-  createTestServers,
-  createRootWithCorePlugins,
-  // TODO: how to resolve this? Where to place this script?
-  // eslint-disable-next-line @kbn/imports/no_boundary_crossing
-} from '@kbn/core-test-helpers-kbn-server';
+// eslint-disable-next-line @kbn/imports/no_boundary_crossing
```

Comment:
Did you find a reason behind the boundary crossing rule?   I'm not familiar with it and hoping to learn more.

Reply:
so, this is firing because the utilities we use here are in packages that have `*test*` in their names, thus are categorized as test packages - this very CLI is categorized as `non-package`, hence the mismatch. 

I can now track and understand where this error is coming from, but I can't reason for or against its correctness in this case. 

```
"tests or mocks" code can not be imported from "non-package" code. Suggestions:
 - Remove the import statement.
 - Limit your imports to "non-package", "server package", "browser package", "common package" or "static" code.
 - Covert to a type-only import.
 - Reach out to #kibana-operations for help.
```